scripts/setup-test-registry

```diff
@@ -45,52 +44,47 @@ mirror_local_image() {
   docker rmi $2
 }
 
+setup_parallel_pull_image() {
+  local image_name="127.0.0.1:51670/$1:parallel-pull-fts"
+  echo "Generating parallel pull image $image_name"
+ 
+  # Get new random bits every time we build 
+  head -c 25m < /dev/urandom > docker-context/random-bits
+  docker build -t "$image_name" docker-context
+
+  echo "Done generating parallel pull image $image_name"
+  mirror_local_image "$image_name" "$image_name"
+}
+
+
 for image in "amazon/amazon-ecs-netkitten" "amazon/amazon-ecs-volumes-test" "amazon/amazon-ecs-pid-namespace-test" "amazon/amazon-ecs-ipc-namespace-test" "amazon/squid" "amazon/awscli" "amazon/image-cleanup-test-image1" "amazon/image-cleanup-test-image2" "amazon/image-cleanup-test-image3" "amazon/fluentd" "amazon/amazon-ecs-agent-introspection-validator" "amazon/amazon-ecs-taskmetadata-validator" "amazon/amazon-ecs-v3-task-endpoint-validator" "amazon/amazon-ecs-container-metadata-file-validator" "amazon/amazon-ecs-elastic-inference-validator" "amazon/amazon-appmesh-plugin-validator"; do
   mirror_local_image "${image}:make" "127.0.0.1:51670/${image}:latest"
 done
 
 
 # Remove the tag so this image can be deleted successfully in the docker image cleanup integ tests
-docker rmi amazon/image-cleanup-test-image1:make
-docker rmi amazon/image-cleanup-test-image2:make
-docker rmi amazon/image-cleanup-test-image3:make
-
-BUSYBOX_IMAGE="busybox:1.29.3"
-NGINX_IMAGE="nginx:1.15.8"
-PYTHON2_IMAGE="python:2.7.15"
-UBUNTU_IMAGE="ubuntu:16.04"
-
-PARALLEL_PULL_FTS_BUSYBOX=${BUSYBOX_IMAGE}
-PARALLEL_PULL_FTS_UBUNTU=${UBUNTU_IMAGE}
-PARALLEL_PULL_FTS_NGINX=${NGINX_IMAGE}
-PARALLEL_PULL_FTS_CONSUL="consul:1.3.0"
-PARALLEL_PULL_FTS_DEBIAN="debian:stable"
-
-BUILD_PLATFORM=$(uname -m)
+docker rmi amazon/image-cleanup-test-image1:make amazon/image-cleanup-test-image2:make amazon/image-cleanup-test-image3:make
+
+mirror_image busybox:1.29.3 "127.0.0.1:51670/busybox:latest"
+mirror_image "$NGINX_IMAGE" "127.0.0.1:51670/nginx:latest"
+mirror_image ubuntu:16.04 "127.0.0.1:51670/ubuntu:latest"
+
+# create a context folder used by docker build. It will only have a file
+# full of random bits so that the parallel pull images are different.
+mkdir -p docker-context
+cat << EOF > docker-context/Dockerfile
+FROM amazon/amazon-ecs-pause:0.1.0 
+ADD random-bits /random-bits
```

Comment:
Why we are doing this?

Reply:
i think so we get distinct image layers aka distinct images for the pulls 🤔

Reply:
Its so we get distinct image layers aka distinct images for the pulls. :)